fv3gfs-physics/fv3gfs/physics/physics_state.py

```diff
@@ -1,120 +1,261 @@
+            "units": "kg/kg",
+            "intent": "inout",
+            "full_model_var": False,
+        }
+    )
+    qrain_t1: FloatField = field(
+        metadata={
+            "name": "physics_rain_mixing_ratio",
+            "units": "kg/kg",
+            "intent": "inout",
+            "full_model_var": False,
+        }
+    )
+    qsnow_t1: FloatField = field(
+        metadata={
+            "name": "physics_snow_mixing_ratio",
+            "units": "kg/kg",
+            "intent": "inout",
+            "full_model_var": False,
+        }
+    )
+    qgraupel_t1: FloatField = field(
+        metadata={
+            "name": "physics_graupel_mixing_ratio",
+            "units": "kg/kg",
+            "intent": "inout",
+            "full_model_var": False,
+        }
+    )
+    qcld_t1: FloatField = field(
+        metadata={
+            "name": "physics_cloud_fraction",
+            "units": "",
+            "intent": "inout",
+            "full_model_var": False,
+        }
+    )
+    pt_t1: FloatField = field(
+        metadata={
+            "name": "physics_air_temperature",
+            "units": "degK",
+            "intent": "inout",
+            "full_model_var": False,
+        }
+    )
+    ua_t1: FloatField = field(
+        metadata={
+            "name": "physics_eastward_wind",
+            "units": "m/s",
+            "intent": "inout",
+            "full_model_var": False,
+        }
+    )
+    va_t1: FloatField = field(
+        metadata={
+            "name": "physics_northward_wind",
+            "units": "m/s",
+            "full_model_var": False,
+        }
+    )
+    delprsi: FloatField = field(
+        metadata={
+            "name": "model_level_pressure_thickness_in_physics",
+            "units": "Pa",
+            "full_model_var": False,
+        }
+    )
+    phii: FloatField = field(
+        metadata={
+            "name": "interface_geopotential_height",
+            "units": "m",
+            "full_model_var": False,
+        }
+    )
+    phil: FloatField = field(
+        metadata={
+            "name": "layer_geopotential_height",
+            "units": "m",
+            "full_model_var": False,
+        }
+    )
+    dz: FloatField = field(
+        metadata={
+            "name": "geopotential_height_thickness",
+            "units": "m",
+            "full_model_var": False,
+        }
+    )
+    wmp: FloatField = field(
+        metadata={
+            "name": "layer_mean_vertical_velocity_microph",
+            "units": "m/s",
+            "full_model_var": False,
+        }
+    )
+    prsi: FloatField = field(
+        metadata={"name": "interface_pressure", "units": "Pa", "full_model_var": False}
+    )
+    prsik: FloatField = field(
+        metadata={
+            "name": "log interface_pressure",
+            "units": "Pa",
+            "full_model_var": False,
+        }
+    )
+    quantity_factory: InitVar[pace.util.QuantityFactory]
```

Comment:
Can you remove the `quantity_factory` attribute? It doesn't get used in this PR, and it can change the meaning of the PhysicsState fairly significantly (from a container of data, to a container of data and an operator).

Reply:
OK thanks, I didn't realize this only provides an arg to post_init. Very cool! We could maybe follow this same pattern in DycoreState for initializing temporary quantities/storages.

Is there ever a case where the user would want to set `tendency` in MicrophysicsState themselves? Are there any other storages here that are only temporaries/outputs (no input is read) that could also be initialized using the quantity_factory (possibly in another PR)?